### ISSUE TITLE
Network quantization

### ISSUE THREAD

**Issue by Jvafaei** [keras-team/keras#13103]
Hi guys,
Right now I'm working on LSTM (specifically language model application) 
My project is to implement Incremental Network Quantization(INQ as an abbreviation in literature) 
Here, it is how it works:
1- Network is fully trained in floating-point format.
2- Then a group of weight(for example 10 percent in each layer) is chosen based on their magnitude (or maybe cost function gradient according to the weight)
3- The selected weights are quantized based on the format given (e.g. Q1.4 ) 
4- The network is trained while the selected weights are freezed to the quantized values (non_selected weights are updating to compensate for the error of quantization)
5- This process continues until the remaining weights are quantized.

Based on the given material, this is what I need:
At the end of each epoch, for each layer, a group of weights is selected by using Keras **_callBack_**  argument in **_fit_** function 
Or maybe it is better to implement it by using **_weight constrain_**  argument in layer function
I don't know how to implement it in Keras

**Comment by gowtham-kp** [comment]
@Jvafaei Please refer to this [documentation](https://keras.io/layers/writing-your-own-keras-layers/) where its clearly mentioned how to implement your own neural network using Keras. Also please go though the source code if necessary. 

Also please note that we only accept issues related to bug/enhancement/feature request.
To ask questions, please see the following resources :
[StackOverflow](https://stackoverflow.com/questions/tagged/keras)
[Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/)
[Keras.io support](https://keras.io/#support)
[Gitter](https://gitter.im/Keras-io/Lobby)
Thanks!
If you think we made a mistake, please open another issue explaining your request in detail
